src/core_plugins/kibana/public/dashboard/directives/grid.js

```diff
@@ -119,91 +138,53 @@ app.directive('dashboardGrid', function ($compile, Notifier) {
       // return the panel object for an element.
       //
       // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-      // ALWAYS CALL makePanelSerializeable AFTER YOU ARE DONE WITH IT
+      // ALWAYS CALL PanelUtils.makeSerializeable AFTER YOU ARE DONE WITH IT
       // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
       function getPanelFor(el) {
         const $panel = el.jquery ? el : $(el);
         const panel = $panel.data('panel');
-
         panel.$el = $panel;
-        panel.$scope = $panel.data('$scope');
-
         return panel;
       }
 
-      // since the $el and $scope are circular structures, they need to be
-      // removed from panel before it can be serialized (we also wouldn't
-      // want them to show up in the url)
-      function makePanelSerializeable(panel) {
-        delete panel.$el;
-        delete panel.$scope;
-      }
-
       // tell gridster to remove the panel, and cleanup our metadata
       function removePanel(panel, silent) {
         // remove from grister 'silently' (don't reorganize after)
         gridster.remove_widget(panel.$el, silent);
-
-        // destroy the scope
-        panel.$scope.$destroy();
-
         panel.$el.removeData('panel');
-        panel.$el.removeData('$scope');
       }
 
       // tell gridster to add the panel, and create additional meatadata like $scope
       function addPanel(panel) {
-        _.defaults(panel, {
-          size_x: 3,
-          size_y: 2
-        });
-
-        // ignore panels that don't have vis id's
-        if (!panel.id) {
-          // In the interest of backwards compat
-          if (panel.visId) {
-            panel.id = panel.visId;
-            panel.type = 'visualization';
-            delete panel.visId;
-          } else {
-            throw new Error('missing object id on panel');
-          }
-        }
-
-        panel.$scope = $scope.$new();
-        panel.$scope.panel = panel;
-        panel.$scope.parentUiState = $scope.uiState;
+        PanelUtils.initializeDefaults(panel);
 
-        panel.$el = $compile('<li><dashboard-panel></li>')(panel.$scope);
+        const panelHtml = `
+            <li>
+                <dashboard-panel remove="removePanelFromState(${panel.panelId})"
+                                 panel="getPanelByPanelId(${panel.panelId})"
+                                 isFullScreenMode="!chrome.getVisible()"
```

Comment:
Dang, nice catch! Wish the linter would catch that, that's the second time I've forgotten about that rule.